pkg/syncer/cnsoperator/manager/init.go

```diff
@@ -61,8 +61,6 @@ type cnsOperator struct {
 // InitCnsOperator initializes the Cns Operator
 func InitCnsOperator(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor,
 	configInfo *commonconfig.ConfigurationInfo, coInitParams *interface{}) error {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
 	ctx = logger.NewContextWithLogger(ctx)
```

Comment:
Can we use `GetNewContextWithLogger` here?

Reply:
And can we rename `NewContextWithLogger` to `NewContext` as this method returns only the context?

Reply:
>  Can we use GetNewContextWithLogger here?

No. If we use GetNewContextWithLogger(), it creates a new background context internally. We should use the existing context supplied with `InitCnsOperator`. 

```golang
// GetNewContextWithLogger creates a new context with context UUID and logger
// set func returns both context and logger to the caller.
func GetNewContextWithLogger() (context.Context, *zap.SugaredLogger) {
	newCtx := NewContextWithLogger(context.Background())
	return newCtx, GetLogger(newCtx)
}

// NewContextWithLogger returns a new child context with context UUID set
// using key CtxId.
func NewContextWithLogger(ctx context.Context) context.Context {
	newCtx := withFields(ctx, zap.String(LogCtxIDKey, uuid.New().String()))
	return newCtx
}

```

Reply:
>  And can we rename NewContextWithLogger to NewContext as this method returns only the context?

We can refactor 
- `NewContextWithLogger` to `NewContext` and 
- `GetNewContextWithLogger` to `NewContextWithLogger` 

I can make this change in a separate PR as this will require updating a lot of files.